src/Lorentz/Contracts/BaseDAO/Types.hs

```diff
@@ -379,6 +397,131 @@ type HasFuncContext s store =
 
 type Entrypoint' cp st s = (cp : st : s) :-> (([Operation], st) : s)
 
+------------------------------------------------------------------------
+-- Permissions
+------------------------------------------------------------------------
+
+newtype Nonce = Nonce { unNonce :: Natural }
+  deriving stock (Generic, Show)
+  deriving newtype (IsoValue, HasAnnotation)
+
+instance TypeHasDoc Nonce where
+  typeDocMdDescription =
+    "Contract-local nonce used to make some data unique."
+
+-- | When singing something, you usually want to use this wrapper over your data.
+-- It ensures that replay attack is not possible.
+data DataToSign d = DataToSign
+  { dsChainId :: ChainId
+  , dsContract :: Address
+  , dsNonce :: Nonce
+  , dsData :: d
+  } deriving stock (Generic)
+    deriving anyclass (IsoValue, HasAnnotation)
```

Comment:
Why do you use `anyclass` deriving here, but `baseDaoAnnOptions` for `Permit` below?

Reply:
I thought that there is no benefit in annotating it, but probably there is nothing bad too.

Reply:
AFAIU it will be annotated in any case, the difference is snake_case vs camlCase.

Reply:
Ah, mmm, now that does not sound like a good default :thinking: :thinking:

Reply:
Why is it not a good default? We want to annotate as much as possible for clarity to make it easier to understand storage/parameter types by looking at Michelson code and using field names sounds like the most natural way to generate annotations.